crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -333,3 +371,148 @@ impl SlidingSyncRoom {
         Default::default()
     }
 }
+
+/// Sliding-Sync extension configuration.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct ExtensionsConfig {
+    /// Request to devices messages with the given config.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_device: Option<ToDeviceConfig>,
+
+    /// Configure the end-to-end-encryption extension.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub e2ee: Option<E2EEConfig>,
+
+    /// Configure the account data extension.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub account_data: Option<AccountDataConfig>,
+
+    /// Extensions may add further fields to the list.
+    #[serde(flatten)]
+    other: BTreeMap<String, serde_json::Value>,
+}
+
+impl ExtensionsConfig {
+    fn is_empty(&self) -> bool {
+        self.to_device.is_none()
+            && self.e2ee.is_none()
+            && self.account_data.is_none()
+            && self.other.is_empty()
+    }
+}
+
+/// Extensions specific response data.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct Extensions {
+    /// To-device extension in response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_device: Option<ToDevice>,
+
+    /// E2EE extension in response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub e2ee: Option<E2EE>,
+
+    /// Account data extension in response.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub account_data: Option<AccountData>,
+}
+
+/// To-device messages extension configuration.
+///
+/// According to [MSC3885](https://github.com/matrix-org/matrix-spec-proposals/pull/3885).
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct ToDeviceConfig {
+    /// Activate or deactivate this extension. Sticky.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub enabled: Option<bool>,
```

Comment:
Same question here.

Reply:
Yes, for `enabled` for all extensions there is semantic difference - as they are 'sticky': Not present means 'no change', `true` enables the extension, `false` disables at.